### PR TITLE
feat(app): implement oauth-buttons directive

### PR DESCRIPTION
**Pull request by kingcody**
Changes:
* Move `oauth-button` to directive and provide spec/e2e test
* ~~`<% does() %>` now accepts yeoman escaped template strings~~
* Implement `bootstrap-social` when `filters.bootstrap == true`

### CODE REVIEW

**Comment by kingcody** [comment]
Requires #1161, #1162, and a rebase.

File: app/templates/client/.jshintrc [hunk 1]
```diff
@@ -22,7 +22,8 @@
     "console": true,
     "$": true,
     "_": true,
-    "moment": true,
+    "moment": true,<% if (filters.jasmine) { %>
+    "jasmine": true,<% } %>
```
Comment (Awk34) [inline app/templates/client/.jshintrc]: This should be in a separate commit

File: app/templates/server/.jshintrc-spec [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
   "extends": ".jshintrc",
-  "globals": {
+  "globals": {<% if (filters.jasmine) { %>
+    "jasmine": true,<% } %>
```
Comment (Awk34) [inline app/templates/server/.jshintrc-spec]: separate commit

**Comment by Awk34** [comment]
I took the time to go back through this. Apart from moving those two JSHint things to a separate commit, it LGTM. Nice work with all the tests for it too.

**Comment by kingcody** [comment]
`.jshintrc` fix is now a separate (fix) commit.
